client/client.go

```diff
@@ -348,14 +349,38 @@ func loadServerTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
 	return ctx, err
 }
 
+func loadPrivateKey(conf *Config) (openssl.PrivateKey, error) {
```

Comment:
Thought:
Should the API surface be narrowed to only the parameters used or not? :thinking: 
Usually I like it that way, but this time, I'm not really sure it's necessary. Hmm

Reply:
you mean passing of conf? exactly what I thought about when I was writing this. maybe since the call is named "loadPrivateKey" it should get privatekey and sslengine.

Reply:
That was my initial thought, when looking at this at least